### PR TITLE
Replacing Travis CI with GitHub actions

### PR DESCRIPTION
**Pull request by CharmisG**
Replacing Travis CI with GitHub actions

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: react-native-square-reader-sdk
```
Comment (hukid) [inline .github/workflows/main.yml:1]: Can you please configure this repo to disable the travis CI check?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+on: [push]
+jobs:
+  install-and-test:
+    runs-on: ubuntu-latest
```
Comment (hukid) [inline .github/workflows/main.yml:5]: It doesn't seem this change works with github action, can you please check?

File: .github/workflows/main.yml (hunk 1 above)
Comment (CharmisG) [reply .github/workflows/main.yml:1]: Removed !!

File: .github/workflows/github-actions-demo.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+name: GitHub Actions Demo
+run-name: ${{ github.actor }} is testing out GitHub Actions 🚀
```
Comment (hukid) [inline .github/workflows/github-actions-demo.yml:2]: Why do we have two workflows?

File: .github/workflows/github-actions-demo.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+on: [push]
+jobs:
+  Explore-GitHub-Actions:
```
Comment (hukid) [inline .github/workflows/github-actions-demo.yml:5]: The job name should be changed to be a meaningful name.

File: reader-sdk-react-native-quickstart/android/app/build.gradle [hunk 5]
```diff
@@ -201,24 +204,28 @@ android {
 
         }
     }
+
+    lintOptions {
+        checkReleaseBuilds false
+        abortOnError false
```
Comment (hukid) [inline reader-sdk-react-native-quickstart/android/app/build.gradle:210]: Why do we ignore errors?

File: reader-sdk-react-native-quickstart/android/app/build.gradle [hunk 6]
```diff
@@ -146,6 +146,9 @@ android {
     compileSdkVersion rootProject.ext.compileSdkVersion
 
     defaultConfig {
+        configurations.all {
+            resolutionStrategy { force 'androidx.core:core-ktx:1.6.0' }
```
Comment (hukid) [inline reader-sdk-react-native-quickstart/android/app/build.gradle:150]: What's the purpose for this line?

File: reader-sdk-react-native-quickstart/android/build.gradle [hunk 7]
```diff
@@ -17,35 +17,48 @@ buildscript {
         afterEvaluate {project ->
             if (project.hasProperty("android")) {
                 android {
-                    compileSdkVersion 31
-                    buildToolsVersion "30.0.3"
+                    compileSdkVersion 32
```
Comment (hukid) [inline reader-sdk-react-native-quickstart/android/build.gradle:20]: Please revert to 31, we don't support 32 yet.

File: reader-sdk-react-native-quickstart/android/gradle.properties [hunk 8]
```diff
@@ -39,13 +39,5 @@ org.gradle.jvmargs=-Xmx4g
 # Automatically convert third-party libraries to use AndroidX
 
 android.enableJetifier=false
-android.jetifier.ignorelist=moshi-1.13.0
-android.jetifier.blacklist = bcprov-jdk15on
 # Version of flipper SDK to use with React Native
 FLIPPER_VERSION=0.99.0
-
-## The Application ID displayed on the Credentials tab in the Square Application Control Panel.
-# SQUARE_READER_SDK_APPLICATION_ID=
```
Comment (hukid) [inline reader-sdk-react-native-quickstart/android/gradle.properties:48]: Do not remove these comments.

File: reader-sdk-react-native-quickstart/android/gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (hukid) [inline reader-sdk-react-native-quickstart/android/gradle/wrapper/gradle-wrapper.properties:3]: Why do we revert to older version?

File: reader-sdk-react-native-quickstart/ios/Podfile [hunk 10]
```diff
@@ -28,5 +27,37 @@ target 'RNReaderSDKSample' do
...
+    end
+
+    # ...but if you bump iOS deployment target, Flipper barfs again "Time.h:52:17: error: typedef redefinition with different types"
+    # We need to make one crude patch to RCT-Folly - set `__IPHONE_10_0` to our iOS target + 1
+    # https://github.com/facebook/flipper/issues/834 - 84 comments and still going...
+    `sed -i -e  $'s/__IPHONE_10_0/__IPHONE_12_0/' #{installer.sandbox.root}/RCT-Folly/folly/portability/Time.h`
   end
-end
+
+  # post_install do |installer|
```
Comment (hukid) [inline reader-sdk-react-native-quickstart/ios/Podfile:55]: please remove these commented code.

File: .github/workflows/github-actions-demo.yml [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (hukid) [inline .github/workflows/github-actions-demo.yml:8]: Should this be v3 as well?

File: .github/workflows/github-actions-demo.yml [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        uses: actions/checkout@v3
+      - name: Install npm dependencies
+        working-directory: ./reader-sdk-react-native-quickstart
+        run: yarn
+      - name: Build
+        working-directory: ./reader-sdk-react-native-quickstart/android
+        env:
+          SQUARE_READER_SDK_APPLICATION_ID: ${{secrets.SQUARE_READER_SDK_APPLICATION_ID}}
+          SQUARE_READER_SDK_REPOSITORY_PASSWORD: ${{secrets.SQUARE_READER_SDK_REPOSITORY_PASSWORD}}
+          TEST_VALUE: abc
```
Comment (hukid) [inline .github/workflows/github-actions-demo.yml:40]: Remove this.

File: .github/workflows/github-actions-demo.yml [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+        run: |
+          echo "Application Id: $SQUARE_READER_SDK_APPLICATION_ID"
```
Comment (hukid) [inline .github/workflows/github-actions-demo.yml:42]: remove these echo cmd.

File: .github/workflows/github-actions-demo.yml [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+        env:
+          SQUARE_READER_SDK_APPLICATION_ID: ${{secrets.SQUARE_READER_SDK_APPLICATION_ID}}
+          SQUARE_READER_SDK_REPOSITORY_PASSWORD: ${{secrets.SQUARE_READER_SDK_REPOSITORY_PASSWORD}}
+      - run: |
+          echo "Application Id: ${{secrets.SQUARE_READER_SDK_APPLICATION_ID}}"
+          sudo arch -x86_64 gem install ffi
+          cd reader-sdk-react-native-quickstart
+          yarn
+          cd ios && arch -x86_64 pod install
+          ruby <(curl https://connect.squareup.com/readersdk-installer) install --app-id ${{secrets.SQUARE_READER_SDK_APPLICATION_ID}} --repo-password ${{secrets.SQUARE_READER_SDK_REPOSITORY_PASSWORD}}
```
Comment (hukid) [inline .github/workflows/github-actions-demo.yml:64]: you are not triggering iOS build.

Comment (hukid) [reply .github/workflows/github-actions-demo.yml:64]: - [ ] add step to actually build iOS project.

**Review by hukid — CHANGES_REQUESTED** [review]

File: .github/workflows/github-actions-demo.yml (hunk 3 above)
Comment (CharmisG) [reply .github/workflows/github-actions-demo.yml]: Removed 2nd one !

File: .github/workflows/github-actions-demo.yml (hunk 4 above)
Comment (CharmisG) [reply .github/workflows/github-actions-demo.yml]: Changed with install-and-test

File: reader-sdk-react-native-quickstart/android/app/build.gradle (hunk 5 above)
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/app/build.gradle]: Removed !!

File: reader-sdk-react-native-quickstart/android/gradle.properties (hunk 8 above)
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/gradle.properties:48]: added !!

File: reader-sdk-react-native-quickstart/android/gradle/wrapper/gradle-wrapper.properties (hunk 9 above)
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/gradle/wrapper/gradle-wrapper.properties]: upgraded version !!

File: reader-sdk-react-native-quickstart/ios/Podfile (hunk 10 above)
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/ios/Podfile]: removed !!

File: .github/workflows/github-actions-demo.yml (hunk 11 above)
Comment (CharmisG) [reply .github/workflows/github-actions-demo.yml]: added as v3 !!

File: .github/workflows/github-actions-demo.yml (hunk 12 above)
Comment (CharmisG) [reply .github/workflows/github-actions-demo.yml]: Removed !

File: .github/workflows/github-actions-demo.yml (hunk 13 above)
Comment (CharmisG) [reply .github/workflows/github-actions-demo.yml]: Removed !!

File: .github/workflows/main.yml [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+      - run: |
+          yarn && yarn lint && yarn test && cd reader-sdk-react-native-quickstart && yarn
```
Comment (hukid) [inline .github/workflows/main.yml:12]: can you please break the line before cd to quickstart folder?

File: reader-sdk-react-native-quickstart/android/gradle.properties [hunk 16]
```diff
@@ -39,13 +39,11 @@ org.gradle.jvmargs=-Xmx4g
...
 # Version of flipper SDK to use with React Native
 FLIPPER_VERSION=0.99.0
 
-## The Application ID displayed on the Credentials tab in the Square Application Control Panel.
-# SQUARE_READER_SDK_APPLICATION_ID=
-
-## The Application Secret from the OAuth tab in the Square Application Control Panel.
-# SQUARE_READER_SDK_REPOSITORY_PASSWORD=
\ No newline at end of file
+### The Application ID displayed on the Credentials tab in the Square Application Control Panel.
```
[the remark this answers is not in the document]
Comment (hukid) [reply reader-sdk-react-native-quickstart/android/gradle.properties:45]: - [ ] revert this change

File: reader-sdk-react-native-quickstart/ios/Podfile [hunk 17]
```diff
@@ -28,5 +27,28 @@ target 'RNReaderSDKSample' do
...
+    installer.aggregate_targets.each do |aggregate_target| 
+      aggregate_target.user_project.native_targets.each do |target|
+        target.build_configurations.each do |config|
+          config.build_settings['LIBRARY_SEARCH_PATHS'] = ['$(SDKROOT)/usr/lib/swift', '$(inherited)']
+        end
+      end
+      aggregate_target.user_project.save
+    end
+
+     # Flipper requires a crude patch to bump up iOS deployment target, or "error: thread-local storage is not supported for the current target"
```
Comment (hukid) [inline reader-sdk-react-native-quickstart/ios/Podfile:41]: nit: remove one extra space.

**Review by hukid — APPROVED** [review]

File: .github/workflows/main.yml [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+        env:
+          SQUARE_READER_SDK_APPLICATION_ID: ${{secrets.SQUARE_READER_SDK_APPLICATION_ID}}
+          SQUARE_READER_SDK_REPOSITORY_PASSWORD: ${{secrets.SQUARE_READER_SDK_REPOSITORY_PASSWORD}}
+      - run: |
+          echo "Application Id: ${{secrets.SQUARE_READER_SDK_APPLICATION_ID}}"
+          sudo arch -x86_64 gem install ffi
+          cd reader-sdk-react-native-quickstart
+          yarn
+          cd ios && arch -x86_64 pod install
+          ruby <(curl https://connect.squareup.com/readersdk-installer) install --app-id ${{secrets.SQUARE_READER_SDK_APPLICATION_ID}} --repo-password ${{secrets.SQUARE_READER_SDK_REPOSITORY_PASSWORD}}
```
[the remark this answers is not in the document]
Comment (CharmisG) [reply .github/workflows/main.yml]: Added !!

File: reader-sdk-react-native-quickstart/android/gradle.properties (hunk 16 above)
[the remark this answers is not in the document]
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/gradle.properties:45]: Done !!

File: .github/workflows/main.yml (hunk 15 above)
Comment (CharmisG) [reply .github/workflows/main.yml]: Done !!

File: reader-sdk-react-native-quickstart/ios/Podfile (hunk 17 above)
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/ios/Podfile:41]: Done !!

**Review by joanc-sq — APPROVED** [review]

**Pull request merged**